### DISCUSSION TITLE
Version 1.1.8

### DISCUSSION THREAD

**Discussion by rainepretorius** [rainepretorius/olarm-ha-integration#23] [Announcements]
**Please take note of the following:**

1. Entity ids of the sensors and alarm panels have changed. 
2. Please redo the setup if you get a  KeyError: 'olarm_arm_code' in your logs, as this new feature needs to be set up. (I apologize for the inconvenience caused, but thought security is the first priority.)

**The following features have been implemented:**
1.  Multiple Olarm device support. (Auto setup from just one API key.)
2.  Added pin support so that if the user wants to secure their device, they can secure it with a pin that needs to be entered for actions.
3. Please note that if you want to use this pin feature, you have to set your pin in the config to something other than 1234567890. Using this value will result in no pin being used/ needed for actioning the alarm.

I apologise for the inconvenience, but I thought adding support for multiple Olarm Devices with only entering one API key is, with the tradeoff of having to fix a few automation and cards, I and hopefully others are willing to make.

**Comment by JTUBB69** [comment]
HI Raine,

Somewhere along the way I seem to have lost connectivity. Having deleted everything and re-installed the latest version, I now get the error below and system is immediately  unavailable.

![image](https://user-images.githubusercontent.com/58702667/230316044-dad5349c-90b7-4fd3-8035-4ff83db288c1.png)


2023-04-06 10:10:00.692 ERROR (MainThread) [custom_components.olarm_sensors.coordinator] Unexpected error fetching Olarm Coordinator (XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX) data: string index out of range
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 250, in _async_refresh
    self.data = await self._async_update_data()
  File "/config/custom_components/olarm_sensors/coordinator.py", line 144, in _async_update_data
    return await self.update_data()
  File "/config/custom_components/olarm_sensors/coordinator.py", line 127, in update_data
    self.pgm_data = await self.api.get_pgm_zones(devices_json)
  File "/config/custom_components/olarm_sensors/olarm_api.py", line 267, in get_pgm_zones
    enabled = pgm_setup[i][0] == "1"
IndexError: string index out of range

**Reply by rainepretorius** [reply]
Hi @JTUBB69,

I will have a look at this asap. I am just writing tests today, so there might be delays in my responses and fixes. I also treated an issue on GitHub and further communication about your problem will be held there.

**Comment by JTUBB69** [comment]
Good luck on tests
